src/frontend/JasmineGraphFrontEnd.cpp

```diff
@@ -1667,6 +1723,190 @@ int JasmineGraphFrontEnd::getRunningHighPriorityTaskCount() {
     return taskCount;
 }
 
+void JasmineGraphServer::pageRank(std::string graphID, double alpha) {
+
+    std::map<std::string, JasmineGraphServer::workerPartition> graphPartitionedHosts =
+            JasmineGraphServer::getWorkerPartitions(graphID);
+    int partition_count = 0;
+    string partition;
+    string host;
+    int port;
+    int dataPort;
+    std::string workerList;
+    Utils utils;
+
+    std::map<std::string, JasmineGraphServer::workerPartition>::iterator workerit;
+    for (workerit = graphPartitionedHosts.begin(); workerit != graphPartitionedHosts.end(); workerit++) {
+        JasmineGraphServer::workerPartition workerPartition = workerit->second;
+        partition = workerPartition.partitionID;
+        host = workerPartition.hostname;
+        port = workerPartition.port;
+        dataPort = workerPartition.dataPort;
+
+        if (host.find('@') != std::string::npos) {
+            host = utils.split(host, '@')[1];
+        }
+
+        workerList.append(host + ":" + std::to_string(port) + ":" + partition + ",");
+    }
+
+    workerList.pop_back();
+    frontend_logger.log("Worker list " + workerList, "error");
+
+    for (workerit = graphPartitionedHosts.begin(); workerit != graphPartitionedHosts.end(); workerit++) {
```

Comment:
Better use the variable name `workerIter` than `workerit` here

Reply:
changed

Reply:
Ok